pkg/logql/engine.go

```diff
@@ -43,11 +44,15 @@ type EngineOpts struct {
 	// MaxLookBackPeriod is the maximum amount of time to look back for log lines.
 	// only used for instant log queries.
 	MaxLookBackPeriod time.Duration `yaml:"max_look_back_period"`
+
+	LogqlOptimizeEnable bool `yaml:"logql_optimize_enable"`
 }
 
 func (opts *EngineOpts) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.DurationVar(&opts.Timeout, prefix+".engine.timeout", 5*time.Minute, "Timeout for query execution.")
 	f.DurationVar(&opts.MaxLookBackPeriod, prefix+".engine.max-lookback-period", 30*time.Second, "The maximum amount of time to look back for log lines. Used only for instant log queries.")
+	f.BoolVar(&opts.LogqlOptimizeEnable, "logql-optimize-enablelog-config-reverse-order", false, "Experimental feature logql optimization")
```

Comment:
```suggestion
	f.BoolVar(&opts.LogqlOptimizeEnable, prefix +".engine.logql-optimize-enable", false, "Experimental feature logql optimization")
```
Can we have same convention for flag like others?